src/Compilers/Core/Portable/Compilation/IExpression.cs

```diff
@@ -0,0 +1,778 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Semantics
+{
+    /// <summary>
+    /// Represents a C# or VB expression.
+    /// </summary>
+    public interface IExpression : IOperation
+    {
+        /// <summary>
+        /// Result type of the expression.
+        /// </summary>
+        ITypeSymbol ResultType { get; }
+        /// <summary>
+        /// If the expression evaluates to a constant value, the value of the expression, and otherwise null.
+        /// </summary>
+        object ConstantValue { get; }
+    }
+
+    /// <summary>
+    /// Represents a C# or VB method invocation.
+    /// </summary>
+    public interface IInvocation : IExpression
```

Comment:
As far as naming is concerned, i think these subinterfaces should include some of their base interface's name.  i.e. we have ExpressionSyntax and BinaryExpressionSyntax.  I think we should have IInvocationExpression here isntead of IInvocation.  This will better match the rest of the compiler API.